dts: x86: intel: alder_lake: Added I2C instances

bindushree98 opened the pull request:
Added I2C instances supported on Alderlake platform

bindushree98 committed:
dts: x86: intel: alder_lake: Added I2C instances

Added I2C instances supported on Alderlake platform

Signed-off-by: Bindu S <bindu.s@intel.com>

MaureenHelm commented on dts/x86/intel/alder_lake.dtsi:
```diff
@@ -8,6 +8,7 @@
 #include <zephyr/dt-bindings/interrupt-controller/intel-ioapic.h>
 #include <zephyr/dt-bindings/i2c/i2c.h>
 #include <zephyr/dt-bindings/pcie/pcie.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
```
Is this needed for adding I2C instances?

MaureenHelm commented on samples/sensor/bme280/boards/intel_adl_crb.conf:
```diff
@@ -0,0 +1,4 @@
+CONFIG_I2C=y
+CONFIG_I2C_DW=y
+
+CONFIG_GPIO=y
```
None of these should be needed

MaureenHelm commented on samples/sensor/bme280/boards/intel_adl_crb.overlay:
```diff
@@ -0,0 +1,13 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&i2c0 {
+        bme280@76 {
+                compatible = "bosch,bme280";
+                reg = <0x76>;
+		status = "okay";
+        };
+};
```
Is this sensor on the board? If it is, this node should be defined in the board dts instead.

bindushree98 replied:
It is not on board sensor, we have connected it to the board and tested it. So, we are not included in board dts.

MaureenHelm commented on dts/x86/intel/alder_lake.dtsi:
```diff
@@ -93,7 +93,7 @@
 
 			status = "disabled";
-                };
+		};
```
This change needs to be squashed into the previous commit

dcpleung commented on boards/x86/intel_adl/intel_adl_crb.yaml:
```diff
@@ -9,6 +9,7 @@ supported:
   - watchdog
   - pwm
   - gpio
+  - i2c
```
TBH, it might be better if you group all of the additions to one PR. Or else you will be doing rebasing for all of these constantly.

bindushree98 committed:
samples: sensor: Enable bme280 sensor using I2C driver for alder_lake

Added overlay and conf to enable bme280 sensor
using i2c on intel_adl_crb board.

Signed-off-by: Bindu S <bindu.s@intel.com>
